### ISSUE TITLE
ContentTypeRange#matches() throws MatchError

### ISSUE THREAD

**Issue by evis** [akka/akka-http#1486]
Code to reproduce bug:

```
import akka.http.scaladsl.model._, ContentType.WithMissingCharset

val range = ContentTypeRange(MediaRange(MediaType.applicationWithOpenCharset("json")))
val mt = WithMissingCharset(MediaType.customWithOpenCharset("some", "type"))
range.matches(mt) // throws MatchError
```

`matches()` is partial function only for `ContextType.Binary` and `ContextType.NonBinary` instances:

```
def matches(contentType: jm.ContentType) =
  contentType match {
    case ContentType.Binary(mt)   ⇒ mediaRange.matches(mt)
    case x: ContentType.NonBinary ⇒ mediaRange.matches(x.mediaType) && charsetRange.matches(x.charset)
  }
```

 But `WithMissingCharset` trait extends base `ContextType`. We should cover this case too.

**Comment by evis** [comment]
Sorry guys, seems like a duplicate of https://github.com/akka/akka-http/pull/1141, hence I close it

**Comment by raboof** [comment]
This should be fixed in 10.0.9 by #1223 - can you confirm that matches your experience?

**Comment by evis** [comment]
Yes, exactly.
